src/Sentry/Envelopes/Envelope.cs

```diff
@@ -132,12 +132,7 @@ public async Task SerializeAsync(Stream stream, IDiagnosticLogger? logger, Cance
                     }
                     catch (Exception exception)
                     {
-                        if (logger is null)
-                        {
-                            throw;
-                        }
-
-                        logger.LogError("Failed to add attachment: {0}.", exception, attachment.FileName);
+                        logger?.LogError("Failed to add attachment: {0}.", exception, attachment.FileName);
```

Comment:
Might be too picky but shouldn't this be part of a separate PR? That way we could merge without Unity blocking the Directory.Build.Targets change.